valhalla/userrequests/duration_utils.py

```diff
@@ -91,7 +91,18 @@ def get_time_allocation_key(telescope_class, proposal_id, min_window_time, max_w
     return TimeAllocationKey(time_allocation.semester.id, telescope_class)
 
 
-def get_total_duration_dict(operator, durations):
+def get_total_duration_dict(operator, proposal_id, requests):
```

Comment:
This kind of looks like you're just passing in a UserRequest, would this be better with a single parameter (userrequest_dict)?